### PR TITLE
fix(app): #9 Resolve issue with scroll event and improve infinite loading

### LINKED ISSUE #9: Infinite Scroll
Check infinite scroll behavior, currently not loading on the desktop. You need to check the scroll event.

### PR DESCRIPTION
**Pull request by AhsanCommits**
### Pull Request for Issue #9: 
Resolve Scroll Event Issue

### Summary:
This pull request fixes issue #9, where the scroll event didn't activate as expected in certain window sizes. It also improves the infinite loading mechanism by preventing multiple scroll events from firing simultaneously.

### Changes Made:
- Fixed the scroll event issue.
- Improved the infinite loading mechanism.

### Testing Done:
- Tested the changes in various scenarios to ensure the scroll event functions reliably.

### Checklist:
- Thoroughly tested the changes.
- Ready for review and merging.

### Additional Comments:
The code has been tested and is ready for review.

### Closes #9
This pull request closes issue #9.

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -1,195 +1,194 @@
...
-        //Inifinite loading wth scroll event
-        window.addEventListener('scroll', () => {
-            console.log("scrolled", window.scrollY) //scrolled from top
-            console.log(window.innerHeight) //visible part of screen
-
-            if (window.scrollY + window.innerHeight >= document.documentElement.scrollHeight) {
-                giphyObj.updateOffset();
-                //update css
-                let list = document.querySelector("#gifs-list");
-                list.style.height = `${giphyObj.offset*10}vh`;
```
Comment (lcrojano) [inline scripts/scripts.js:172]: Keeping the height of the list item in infinite scrolling is needed. It allows the page to dynamically extend its visible area, accommodating more content as users scroll, preventing layout issues. Without it, new content may overlap or appear off-screen, causing display problems.
Like this
[screen-capture (1).webm](https://github.com/lcrojano/Giphy_Explorer/assets/5177737/a35f0fcf-2fb4-4823-b10d-43551894d2f1)

File: scripts/scripts.js [hunk 2]
```diff
@@ -1,195 +1,194 @@
...
+    loadGifs(giphyObj.trending_api(), updateMainView);
+  }
+
+  // Improved scroll event handling
+  let loading = false; // Prevent multiple scroll events
+
+  window.addEventListener('scroll', () => {
+    if (
+      !loading &&
+      window.innerHeight + window.scrollY >= document.body.offsetHeight - 200
```
Comment (lcrojano) [inline scripts/scripts.js:165]: window.innerHeight + window.scrollY >= document.body.offsetHeight - 1

could also work

**Review by lcrojano — CHANGES_REQUESTED** [review]
Great solution! I have a few suggestions:

**Retain Height Adjustment:** It's crucial to keep the height adjustment for the list item, as it's essential for the infinite scroll illusion.
**Optimize Offset:** Consider using a smaller offset, such as 1, instead of 200.

These changes will effectively address the issue, and your work is greatly appreciated! Thank you for your efforts.

**Comment by AhsanCommits** [comment]
Thank you for the suggestions! I've made the following updates:

### 1. Retained Height Adjustment:
To maintain the infinite scroll illusion.
### 2. Optimized Offset:
Set it to 1 for improved scrolling.

Your feedback is greatly appreciated, and the code has been adjusted accordingly. Please review the changes, and feel free to share any further input.

**Pull request merged**